docs/data_model.md

```diff
@@ -0,0 +1,405 @@
+        uuid lead_provider_id
+        uuid user_id
+        uuid schedule_id
+        uuid itt_provider_id
+        uuid school_id
+        uuid private_childcare_provider_id
+        string employer_name
+        string employment_role
+        string funding_choice
+        string headteacher_status
+        string ineligible_for_funding_reason
+        string school_urn
+        string lead_provider_approval_status
+        string training_status
+        boolean works_in_school
+        boolean eligible_for_funding
+        boolean targeted_delivery_funding_eligibility
+        string teacher_catchment
+        string teacher_catchment_iso_country_code
+        string teacher_catchment_country
+        boolean lead_mentor
+        datetime accepted_at
+        datetime updated_at
+    }
+
+    Cohort {
+        uuid id
+        integer start_year
+    }
+
+    ParticipantOutcome {
+        uuid id
+        string state
+        date completion_date
+        uuid declaration_id
+        datetime created_at
+    }
+
+    Declaration {
+        uuid id
+        uuid application_id
+        string state
+        string declaration_type
+        date declaration_date
+        datetime updated_at
+    }
+
+    School {
+        uuid id
+        string urn
+        string ukprn
+    }
+
+    Statement {
+        uuid id
+        enum month
+        integer year
+        date deadline_date
+        uuid cohort_id
+        uuid lead_provider_id
+        datetime marked_as_paid_at
+        decimal reconcile_amount
+        string state
+    }
+
+    StatementItem {
+        uuid id
+        uuid statement_id
+        uuid declaration_id
+        string state
+    }
+
+    Contract {
+        uuid statement_id
+        uuid course_id
+        uuid contract_template_id
+    }
+
+    ContractTemplate {
+        boolean special_course
+        decimal recruitment_target
+        decimal per_participant
+        decimal output_payment_percentage
+        decimal number_of_payment_periods
+        decimal service_fee_percentage
+        decimal service_fee_installments
+    }
+
+    CourseGroup {
+        uuid id
+        string name
+    }
+
+    Schedule {
+        uuid id
+        uuid course_group_id
+        uuid cohort_id
+        string name
+        date declaration_starts_on
+        date schedule_applies_from
+        date schedule_applies_to
+        string declaration_type
+    }
+
+    ParticipantIdChange {
+        uuid id
+        uuid user_id
+        uuid from_participant_id
+        uuid to_participant_id
+    }
+
+    ApiToken {
+        uuid id
+        uuid lead_provider_id
+        string hashed_token
+        datetime last_used_at
+    }
+```
+
+### Users and Applications
+
+NPQ is a much simpler process in comparisson with ECF. An application is made once by the teacher themselves. We don't need to worry about ownership or restricting access via the API, so we allow applications to `belong_to` a user directly.
```

Comment:
not sure I understand the restriction here? can you elaborate more about what we're restricting? Also I wonder if ECF should be mentioned here? should we look at this document as an explanation of the model rather than comparison with ECF 🤔